eng/MultiTargetRoslynComponent.targets.template

```diff
@@ -0,0 +1,31 @@
+<Project>
+  <Target Name="_{TargetPrefix}GatherAnalyzers">
+
+    <ItemGroup>
+      <_{TargetPrefix}Analyzer Include="@(Analyzer)" Condition="'%(Analyzer.NuGetPackageId)' == '{NuGetPackageId}'" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_{TargetPrefix}AnalyzerMultiTargeting" 
+          Condition="'$(SupportsRoslynComponentVersioning)' != 'true'" 
+          AfterTargets="ResolvePackageDependenciesForBuild;ResolveNuGetPackageAssets"
+          DependsOnTargets="_{TargetPrefix}GatherAnalyzers">
+
+    <ItemGroup>
+      <!-- Remove our analyzers targeting roslyn4.x -->
+      <Analyzer Remove="@(_{TargetPrefix}Analyzer)"
+                Condition="$([System.String]::Copy('%(_{TargetPrefix}Analyzer.Identity)').IndexOf('roslyn4')) &gt;= 0"/>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_{TargetPrefix}RemoveAnalyzers" 
+          Condition="'$({DisableSourceGeneratorPropertyName})' == 'true'"
+          AfterTargets="ResolvePackageDependenciesForBuild;ResolveNuGetPackageAssets"
+          DependsOnTargets="_{TargetPrefix}GatherAnalyzers">
+
+    <!-- Remove all our analyzers -->
```

Comment:
Not blocking, but one clarification of the way this works is only for this package's analyzers, it does not have any effect if the analyzer from the shared framework would win.  It might make sense to put package in the name of the property but that gets a little long.  I guess if we document them we should make it clear it only applies to package.

Reply:
> it does not have any effect if the analyzer from the shared framework would win

For the analyzers in the shared fx ref-pack, I was relying on https://github.com/dotnet/roslyn/issues/55518 to provide this functionality, if needed.

I wouldn't object to putting `Package` in the property name, if we think that is the best option.

`DisableSystemTextJsonSourceGenerator` => `DisableSystemTextJsonPackageSourceGenerator`

It expresses intent.

Do you have a strong opinion either way?